src/libfetchers/git-utils.cc

```diff
@@ -57,6 +58,81 @@ bool operator == (const git_oid & oid1, const git_oid & oid2)
     return git_oid_equal(&oid1, &oid2);
 }
 
+namespace {
+
+int matchesDotPlusGit(const std::string& str) {
+    // String must have at least 4 characters (at least one '.' plus "git")
+    if (str.size() < 4) {
+        return 0;
+    }
+
+    // Count consecutive dots at the beginning
+    size_t dotCount = 0;
+    while (dotCount < str.size() && str[dotCount] == '.') {
+        dotCount++;
+    }
+
+    // Must have at least one dot
+    if (dotCount == 0) {
+        return 0;
+    }
+
+    // After the dots, check if the remaining string is exactly "git"
+    if ((str.size() == dotCount + 3) &&
+           (str[dotCount] == 'g') &&
+           (str[dotCount + 1] == 'i') &&
+           (str[dotCount + 2] == 't')) {
+        return dotCount;
+    }
+    return 0;
+}
+
+std::string escapeDotGit(const std::string& filename) {
```

Comment:
TODO: We should only do this for the tarball cache.

Reply:
Also we need to bump the version of this cache as older version might otherwise be confused.